typing/env.mli

```diff
@@ -518,8 +518,8 @@ val reset_cache: preserve_persistent_env:bool -> unit
 val reset_cache_toplevel: unit -> unit
 
 (* Remember the name of the current compilation unit. *)
-val set_unit_name: Compilation_unit.t option -> unit
-val get_unit_name: unit -> Compilation_unit.t option
+val set_unit_name: Unit_info.t option -> unit
+val get_unit_name: unit -> Unit_info.t option
```

Comment:
Is there a reason here why you don't stick closer to the upstream signature ? 
Having a specialized `get_unit_name` that returns the `Compilation_unit.t` directly might reduce the diff a bit ?

Upstream signature is:

```ocaml
(* Remember the current compilation unit. *)
val set_current_unit: Unit_info.t -> unit
val get_current_unit : unit -> Unit_info.t option
val get_current_unit_name: unit -> string (* Compilation_unit.t *)
```